plugins/modules/idrac_system_erase.py

```diff
@@ -0,0 +1,569 @@
+JOB_FILTER = "Jobs?$expand=*($levels=1)"
+
+ERASE_SUCCESS_COMPLETION_MSG = "Successfully completed the system erase operation."
+ERASE_SUCCESS_SCHEDULED_MSG = "Successfully submitted the job for system erase operation."
+ERASE_SUCCESS_POWER_ON_MSG = "Successfully completed the system erase operation and powered on " \
+                             "the server."
+NO_COMPONENT_MATCH = "Unable to complete the operation because the value entered for the " \
+                     "'component' is not in the list of acceptable values."
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The value for the 'job_wait_timeout' parameter cannot be " \
+                               "negative or zero."
+WAIT_TIMEOUT_MSG = "The job is not complete after {0} seconds."
+INVALID_COMPONENT_WARN_MSG = "Erase operation is not performed on these components - " \
+                             "{unmatching_components_str_format} as they are either invalid or " \
+                             "inapplicable."
+FAILURE_MSG = "Unable to complete the system erase operation."
+CHANGES_FOUND_MSG = "Changes found to be applied."
+NO_CHANGES_FOUND_MSG = "No changes found to be applied."
+
+
+class SystemErase():
+    def __init__(self, idrac, module):
+        """
+        Initializes the class instance with the provided idrac and module parameters.
+
+        :param idrac: The idrac parameter.
+        :type idrac: Any
+        :param module: The module parameter.
+        :type module: Any
+        """
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self):
+        """
+        Executes the function with the given module.
+
+        :param module: The module to execute.
+        :type module: Any
+        :return: None
+        """
+
+    def get_system_erase_url(self):
+        """
+        Retrieves the URL for the system erase operation.
+
+        Returns:
+            str: The URL for the system erase operation.
+        """
+        url = self.get_url()
+        resp = get_dynamic_uri(self.idrac, url)
+        system_erase_url = (
+            resp.get('Links', {})
+            .get(OEM, {})
+            .get(MANUFACTURER, {})
+            .get(LC_SERVICE, {})
+            .get(ODATA, {})
+        )
+        return system_erase_url
+
+    def get_url(self):
+        """
+        Retrieves the URL for the resource.
+
+        Returns:
+            str: The URL for the resource.
+
+        Raises:
+            AnsibleExitJson: If the resource ID is not found.
+        """
+        res_id = self.module.params.get('resource_id')
+        if res_id:
+            uri = MANAGERS_URI + "/" + res_id
+        else:
+            uri, error_msg = validate_and_get_first_resource_id_uri(
+                self.module, self.idrac, MANAGERS_URI)
+            if error_msg:
+                self.module.exit_json(msg=error_msg, failed=True)
+        return uri
+
+    def get_job_status(self, erase_component_response):
```

Comment:
Can you please check if we have same function in utils if not please move it.

Reply:
Its not available in the utils. Keeping this in the module itself. Will take care of adding to utils in the upcoming release